rs/nervous_system/canisters/src/registry.rs

```diff
@@ -28,6 +29,34 @@ impl RegistryCanister {
             canister_id: REGISTRY_CANISTER_ID,
         }
     }
+
+    async fn execute_with_retries<F, Fut, Response>(
+        &self,
+        max_attempts: u8,
+        call: F,
+    ) -> Result<Response, NervousSystemError>
+    where
+        F: Fn() -> Fut,
+        Fut: Future<Output = Result<Response, CallFailed>>,
+    {
+        for _ in 0..max_attempts {
+            match call().await {
+                Ok(response) => return Ok(response),
+                Err(e) if e.is_immediately_retryable() => {
+                    continue;
+                }
+                Err(e) => {
+                    return Err(NervousSystemError::new_with_message(format!(
+                        "Request failed with error: {e}"
+                    )));
+                }
+            }
+        }
+
+        Err(NervousSystemError::new_with_message(format!(
+            "Request failed after {max_attempts} attempts"
+        )))
+    }
```

Comment:
This is supposed to be abstracted in a library by @oggy-dfin : is this already available for public use?

Reply:
Yes I've just published it at https://crates.io/crates/ic-call-retry - thanks for the ping!

Reply:
@oggy-dfin Any chance this will be available in the monorepo soon?  Does it need to be published under dfinity first?

Reply:
I'm not sure - it doesn't need to be in the monorepo necessarily, but I'm not sure what the policy is regarding pulling in crates from external sources, or if this should be published under the dfinity github account.